vulnerability/worker.go

```diff
@@ -116,6 +118,13 @@ func (f *VulnerabilityWorker) Run(ctx context.Context) {
 			f.log.Info("VulnerabilityWorker.work waiting on workers")
 			f.wg.Wait()
 			f.log.Info("VulnerabilityWorker.work finished waiting on workers")
+
+			err := f.cleaner.CleanSnapshots(ctx, snapshotIds)
```

Comment:
I would try to align with other methods that get called here.
It is worth spending some time to make `CleanSnapshots` run in a goroutine and accept a channel of snapshots to delete.

Reply:
Seems like it makes the cleanup flow complicated.
Decided for the current iteration to clean snapshots at the end of the cycle without aligning with the rest of the steps.